src/main/kotlin/com/evalify/evalifybackend/quiz/service/QuizInstructorService.kt

```
@@ -31,14 +33,27 @@ class QuizInstructorService(val quizRepository: QuizRepository) {
      * @return QuizPreviewDTO object
      */
     private fun mapToQuizPreviewDTO(quiz: Quiz): QuizPreviewDTO {
+        // Initialize lazy collections within the transaction boundary
+        val batchNames = try {
+            quiz.batch.map { it.name }
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)